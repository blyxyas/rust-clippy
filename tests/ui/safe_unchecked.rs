// run-rustfix
#![allow(unused)]
#![warn(clippy::safe_unchecked)]

fn should_lint_unchecked() {}
fn should_not_lint() {} // Name doesn't end with "_unchecked"
unsafe fn should_not_lint_unchecked() {} // It's an unsafe function

trait SomeTrait {
    fn method_should_not_lint_unchecked();
}

impl SomeTrait for usize {
    fn method_should_not_lint_unchecked() {}
}

fn main() {}
