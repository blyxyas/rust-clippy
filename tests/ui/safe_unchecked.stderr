error: this function ending with `_unchecked` isn't marked with `unsafe`
  --> $DIR/safe_unchecked.rs:5:4
   |
LL | fn should_lint_unchecked() {}
   |    ^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `-D clippy::safe-unchecked` implied by `-D warnings`
help: add an unsafe marker
   |
LL | unsafe fn should_lint_unchecked() {}
   | ++++++

error: aborting due to previous error

